Client/SafeBrowsing/GSafeBrowsing.swift

```diff
@@ -0,0 +1,316 @@
+    private let session = URLSession(configuration: .ephemeral)
+    private let database = SafeBrowsingDatabase()
+    
+    public static let shared = SafeBrowsingClient()
+    
+    private init() {
+    }
+    
+    func find(_ hashes: [String], _ completion: @escaping (_ isSafe: Bool, Error?) -> Void) {
+        let discoveredHashes = database.find(hashes)
+        if discoveredHashes.isEmpty {
+            return completion(true, nil)
+        }
+        
+        let clientInfo = ClientInfo(clientId: SafeBrowsingClient.clientId,
+                                    clientVersion: SafeBrowsingClient.version)
+        
+        let threatTypes: [ThreatType] = [.malware,
+                                         .socialEngineering,
+                                         .unwantedSoftware,
+                                         .potentiallyHarmfulApplication]
+        
+        let platformTypes: [PlatformType] = [.any]
+        let threatEntryTypes: [ThreatEntryType] = [.url, .exe]
+        
+        let threatInfo = ThreatInfo(threatTypes: threatTypes,
+                                    platformTypes: platformTypes,
+                                    threatEntryTypes: threatEntryTypes,
+                                    threatEntries: discoveredHashes.map {
+                                        return ThreatEntry(hash: $0, url: nil, digest: nil)
+                                    }
+        )
+        
+        do {
+            let body = FindRequest(client: clientInfo, threatInfo: threatInfo)
+            let request = try encode(.post, endpoint: .fullHashes, body: body)
+            executeRequest(request, type: FindResponse.self) { response, error in
+                if let error = error {
+                    return completion(false, error)
+                }
+                
+                if let response = response {
+                    if !response.matches.isEmpty {
+                        return completion(false, nil)
+                    }
+                }
+                
+                completion(true, nil)
+            }
+        } catch {
+            completion(false, error)
+        }
+    }
+    
+    func fetch(_ completion: @escaping (Error?) -> Void) {
+        let clientInfo = ClientInfo(clientId: SafeBrowsingClient.clientId,
+                                    clientVersion: SafeBrowsingClient.version)
+        
+        let constraints = Constraints(maxUpdateEntries: UInt32(SafeBrowsingClient.maxBandwidth),
+                                      maxDatabaseEntries: UInt32(SafeBrowsingClient.maxDiskSpace),
+                                      region: "US",
+                                      supportedCompressions: [.raw],
+                                      language: nil,
+                                      deviceLocation: nil)
+        
+        let lists = [
+            ListUpdateRequest(threatType: .malware,
+                              platformType: .any,
```

Comment:
You might want to specify iOS here instead of ANY. There's not much point in downloading large lists of Windows malware sites on an iOS device.

Reply:
> You might want to specify iOS here instead of ANY. There's not much point in downloading large lists of Windows malware sites on an iOS device.

If I do that, I'll have a mismatch between the desktop, android, and iOS. All platforms atm pass "any". Why? I have no idea. With iOS 13, users will be able to download any file right from Safari and I guess WebKit too (not sure about web-kit, but we do have a download manager feature coming out soon for Brave-iOS).

So if I change it, won't it open us up to a security problem?

Reply:
On desktop, I'm pretty sure we pass the right platform (Windows, Mac, Linux): https://cs.chromium.org/chromium/src/components/safe_browsing/db/v4_protocol_manager_util.cc?l=112&rcl=ceb3caaecb93bd80f74895e70139d225c8cd84c7

It is expected that there will be differences between platforms because they are not necessarily subject to the same threats, but if an attack works on multiple platforms, then I expect it to be part of the list for each platform.

I can double-check with Google if you'd like, but I think we should start with specifying the correct platform and change this later if we find there's a problem.

Reply:
Fixed. I have changed it to only `iOS`

Reply:
It looks like it was updated for every threat type except for `malware`.